tests/test_input.py

```
@@ -2619,12 +2619,12 @@ def test_write_vasp_files_from_doped_dict(self):
             ],
         }
         with self.assertRaises(ValueError) as e:
-            no_bulk_error = ValueError(
-                "Input `defect_entries` dict matches `doped`/`PyCDT` format, but no 'bulk' entry "
-                "present. Please try again providing a `bulk` entry in `defect_entries`."
-            )
             dist = input.Distortions(vacancies)
-            self.assertIn(no_bulk_error, e.exception)
+        self.assertIn(
```

Ruff (0.11.9)

Use a regular `assert` instead of unittest-style `assertIn`

Replace `assertIn(...)` with `assert ...`

(PT009)